### DISCUSSION TITLE
For multi-step predictions, have you applied this model into the temperature dataset?

### DISCUSSION THREAD

**Discussion by wenlongliao123** [oliverguhr/transformer-time-series-prediction#12] [General]
Hi.Oliver. For one-step predictions,the performance of this model is acceptable. For multi-step predictions, have you applied this model into the temperature dataset?
Today, I apply this model into the temperature dataset for multi-step prediction based on your codes 
However,the accuracy is very bad. After 200 epochs, the forecasting tempture curves is parallel to the X-axis.

**Comment by oliverguhr** [comment]
Hi @wenlongliao123,
yes, I did. Have you tried to set the ```calculate_loss_over_all_values = True```. Depending on the problem setting this flag might help.

However, the multi-step prediction isn't the best idea anyway. One could simply repeatedly use the single-step model. 

If you want to apply transformers to data you should have a look at this repo: https://github.com/AIStream-Peelout/flow-forecast

Best,
Oliver

**Reply by CrazyCyberbug** [reply]
hi Oliver. I'm new to the world of transformers. I tried running your transformer model, both single-step and multi-step variations. Both of these converge very well on the dummy sine wave data although do not converge at all when the real temperature dataset is fed to these models. Both render a line parallel to the x-axis.  I tried setting the `calculate_loss_over_all_values = True`, which slightly conforms to the data's shape but does not account for all the variations in data. 

the only code changes I made to run the  models on the dataset was to uncomment a few of the following lines

   >from pandas import read_csv  #  UNCOMMENTED
   >series = read_csv('daily-min-temperatures.csv', header=0, index_col=0, parse_dates=True, squeeze=True)    #  UNCOMMENTED
   >from sklearn.preprocessing import MinMaxScaler #  UNCOMMENTED
   >scaler = MinMaxScaler(feature_range=(-1, 1)) #  UNCOMMENTED
   >amplitude = scaler.fit_transform(series.to_numpy().reshape(-1, 1)).reshape(-1) #  UNCOMMENTED
   >#amplitude = scaler.fit_transform(amplitude.reshape(-1, 1)).reshape(-1)  #  COMMENTED

Am I missing something .. does the code require any more changes to work with the dataset?. kindly help me out. 

best regards, 
Ananth
